core/os/android/adb/device.go

```diff
@@ -222,39 +227,40 @@ func newDevice(ctx context.Context, serial string, status bind.Status) (*binding
 		d.To.Configuration.AnglePackage = anglePackage
 	}
 
-	// If the VkRenderStagesProducer layer exist, we assume the render stages producer is
-	// implemented in the layer.
-	for _, l := range d.To.Configuration.GetDrivers().GetVulkan().GetLayers() {
-		if l.Name == "VkRenderStagesProducer" {
-			capability := d.To.Configuration.PerfettoCapability
-			if capability == nil {
-				capability = &device.PerfettoCapability{
-					GpuProfiling: &device.GPUProfiling{},
+	// Query infos related to the Vulkan driver
+	if d.To.Configuration.GetDrivers() != nil && d.To.Configuration.GetDrivers().GetVulkan() != nil {
+
+		// If the VkRenderStagesProducer layer exist, we assume the render stages producer is
+		// implemented in the layer.
+		for _, l := range d.To.Configuration.GetDrivers().GetVulkan().GetLayers() {
+			if l.Name == "VkRenderStagesProducer" {
+				capability := d.To.Configuration.PerfettoCapability
+				if capability == nil {
+					capability = &device.PerfettoCapability{
+						GpuProfiling: &device.GPUProfiling{},
+					}
+					d.To.Configuration.PerfettoCapability = capability
```

Comment:
@rschiu you added these checks in https://github.com/google/gapid/pull/3693, but it looks like this line (to update `d.To.Configuration.PerfettoCapability` with the newly created `capability`) was missing?

Let me know what you think.